.travis.yml

```diff
@@ -1,14 +1,27 @@
-# GCE VMs have better performance (will be upgrading to premium VMs soon)
-sudo: required
+dist: xenial
+group: stable
 
 language: scala
-jdk: openjdk8
 
 cache:
   directories:
     - $HOME/.ivy2/cache
     - $HOME/.sbt
     - $HOME/.rvm/
+    - $HOME/.sdkman
+
+before_install:
+  # adding $HOME/.sdkman to cache would create an empty directory, which interferes with the initial installation
+  - "[[ -d /home/travis/.sdkman/ ]] && [[ -d /home/travis/.sdkman/bin/ ]] || rm -rf /home/travis/.sdkman/"
+  - curl -sL https://get.sdkman.io | bash
+  - echo sdkman_auto_answer=true > /home/travis/.sdkman/etc/config
+  - source "/home/travis/.sdkman/bin/sdkman-init.sh"
+
+install:
+  - sdk install java $(sdk list java | grep -o "$ADOPTOPENJDK\.[0-9\.]*hs-adpt" | tail -1)
```

Comment:
Alternatively, we could stick to using `install-jdk.sh` which (apparently, but we should verify) was just updated by Travis CI (https://github.com/sormuras/bach/issues/62), by adding the following as an `install` step:

```yaml
- source install-jdk.sh --url "https://api.adoptopenjdk.net/v2/binary/releases/openjdk$ADOPTOPENJDK?openjdk_impl=hotspot&os=linux&arch=x64&release=latest&heap_size=normal&type=jdk"
```

This is where those URLs redirect to:

```
09:11:36 $ for ADOPTOPENJDK in 8 11; do
> curl -I "https://api.adoptopenjdk.net/v2/binary/releases/openjdk$ADOPTOPENJDK?openjdk_impl=hotspot&os=linux&arch=x64&release=latest&heap_size=normal&type=jdk"
> done
HTTP/2 302
date: Tue, 09 Jul 2019 08:19:37 GMT
content-type: text/plain; charset=utf-8
content-length: 150
set-cookie: __cfduid=dbe707790dae1c34dc2e23ec1cdf42a581562660376; expires=Wed, 08-Jul-20 08:19:36 GMT; path=/; domain=.adoptopenjdk.net; HttpOnly
x-powered-by: Express
access-control-allow-origin: *
location: https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b04/OpenJDK8U-jdk_x64_linux_hotspot_8u212b04.tar.gz
vary: Accept
set-cookie: b7b892882bae631693e1ea44963ef628=afdd0ec2d9aff4e059f50dfa0530835e; path=/; HttpOnly; Secure
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: 4f38dfbb3ff9dbdb-LHR

HTTP/2 302
date: Tue, 09 Jul 2019 08:19:37 GMT
content-type: text/plain; charset=utf-8
content-length: 154
set-cookie: __cfduid=dab5451e461b1bce70eeaab3f5054a6981562660377; expires=Wed, 08-Jul-20 08:19:37 GMT; path=/; domain=.adoptopenjdk.net; HttpOnly
x-powered-by: Express
access-control-allow-origin: *
location: https://github.com/AdoptOpenJDK/openjdk11-binaries/releases/download/jdk-11.0.3%2B7/OpenJDK11U-jdk_x64_linux_hotspot_11.0.3_7.tar.gz
vary: Accept
set-cookie: b7b892882bae631693e1ea44963ef628=afdd0ec2d9aff4e059f50dfa0530835e; path=/; HttpOnly; Secure
expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
server: cloudflare
cf-ray: 4f38dfbf3df9ce2f-LHR
```

Reply:
```yaml
before_install: wget https://github.com/sormuras/bach/raw/master/install-jdk.sh
```

Reply:
+1

If that doesn't work, jabba supports installing from an url.

```
jabba install 1.8-adopt-latest=tgz+'https://api.adoptopenjdk.net/v2/binary/releases/openjdk8?openjdk_impl=hotspot&os=linux&arch=x64&release=latest&heap_size=normal&type=jdk'
```

works on my machine.

Reply:
Since we cache the JDK, we should also check that the cached version is still the latest.